### ISSUE TITLE
GRU+SVM+DROPOUT+LR-DECAY

### ISSUE THREAD

**Issue by mpekalski** [zalandoresearch/fashion-mnist#36]
GRU+SVM+DROPOUT+LR-DECAY in TF

https://github.com/mpekalski/zalando/blob/master/GRU%2BSVM%2BDROPOUT%2BLR-DECAY.ipynb

100 epochs, test accuracy: 0.9841001033782959

No preprocessing.

**Comment by mpekalski** [comment]
@AFAgarap thanks for the great code base :+1:

**Comment by AFAgarap** [comment]
Thank you, @mpekalski!
I hope they'll say the same at the university. :) Nice tweaks, btw. :+1: It's been educational.

**Comment by mpekalski** [comment]
Update:

https://github.com/mpekalski/zalando/blob/master/GRU%2BSVM%2BDROPOUT%2BLR-DECAY-300epochs.ipynb

I run the same model for 300 epochs and got test accuracy of 0.9886001348495483

**Comment by hanxiao** [comment]
:open_mouth:

**Comment by hanxiao** [comment]
@mpekalski @AFAgarap  could you clarify why do we need `epoch_y[epoch_y == 0] = -1`? this network has less capacity (in terms of parameters) than other deep learning baselines in the README. For me it's not very intuitive to understand why it performs much better than all conv-net baselines.

**Comment by mpekalski** [comment]
I think I found an issue:
https://github.com/mpekalski/zalando/blob/master/problem.ipynb

Although I am using TF 1.3.0, compiled two days ago, it does not load the proper dataset, see the link above for more details. It loads original MNIST.
```
from tensorflow.examples.tutorials.mnist import input_data
data = input_data.read_data_sets('data/fashion', one_hot=True)
```

I checked counts of train and test datasets and they are 55k and 10k, respectively.

**Comment by hanxiao** [comment]
did you download fashion-mnist data set (4 files) and put it to ./data/fashion, where "./" should be the directory of problem.ipynb?

**Comment by mpekalski** [comment]
No, I did not. I has downloaded the files automatically and created the folder structure, so I thought it ought to work this way. Apparently those were the original files. Somehow I did not expect them to have exactly the same names. :) 

Anyway, sorry for the issue. I will have to reach those 0.98 with some other model. :P

**Comment by hanxiao** [comment]
You have to download the data first, otherwise Tensorflow will download and use default MNIST data. Now your 0.98 test accuracy makes sense to me, as it is on the original MNIST.

**Comment by hanxiao** [comment]
close this benchmark as it is evaluated on the original MNIST.

**Comment by AFAgarap** [comment]
@hanxiao SVMs typically need the y's in range of `{-1, +1}`. Plus, a study was conducted to prove the effectiveness of SVM over Softmax: http://deeplearning.net/wp-content/uploads/2013/03/dlsvm.pdf. However, in the study, they did not use RNN for the neural network. They used CNN. Hence, my proposed model -- GRU+SVM. 
On the note of having trained on the original MNIST, I was also surprised that it performed that well on your dataset, ~98.8%. Nevertheless, I applaud the tweaks by @mpekalski.
